R/FilterSpectroOut.R

```diff
@@ -0,0 +1,659 @@
+#' to 'TRUE'.
+#' Default is 'Specific'.
+#' @param filterProteotryptic A boolean value, defining if peptides should be
+#' filtered based on if they are proteotypic.
+#' Default is 'FALSE'.
+#' @param infoProteotypic A character string or numeric giving the column name
+#' or column number of \code{annotPP} in which annotation which peptides are
+#' proteotypic is provided.
+#' Default is 'isProteotypic'.
+#' @param nameProteotypic A character string defining proteotypic peptides are
+#' annotated in the \code{infoProteotypic} colunm in \code{annotPP}. Onl
+#' peptides annotated accordingly will remain in the data if
+#' \code{filterProteotryptic} is set to 'TRUE'.
+#' Default is 'True'.
+#' @param filterMissedCleaved A boolean value defining if peptides should be
+#' filtered based on the number of missed cleavages.
+#' Default is 'FALSE'.
+#' @param maxMissedCleave A numeric value, defining maximal number of missed
+#' cleavages allowed per peptide if \code{filterMissedCleaved} is set to 'TRUE'.
+#' Default is '2'.
+#' @param infoMissedCleave A character string or numeric giving the column name
+#' or column number of \code{annotPP} in which number of missed cleavage are
+#' provided.
+#' Default is 'NMissedCleavages'.
+#'
+#' @return Returns a list of matrices containing preprocessed and filtered
+#' quantities of interest(e.g. peptide, modified peptide, precursor) and protein
+#' quantities (OR a single matrix containing preprocessed quantities of interest
+#' if a 'quantityMatrix' was provided. Feature and samples names (i.e., row and
+#' column names) of the different matrices in the list match.
+#'
+#' If a "quantityMatrix' is provided the function returns a single matrix
+#' containing preprocessed quantities of interest.
+#'
+#' Rows represent features and columns samples.
+#'
+#' @export
+preprocessQuantityMatrix <- function(quantityList=NULL, quantityMatrix=NULL,
+                                     annotPP=NULL, annotS=NULL, mode="default",
+                                     logT=TRUE, filterMinLogQuant=TRUE,
+                                     thresholdMinLogQuant=10, filterNA="all",
+                                     maxNAs=0,
+                                     infoCondition="Condition",
+                                     nameIDQuant="quantID",
+                                     nameProtQuant="Protein",
+                                     filterTryptic=TRUE,
+                                     infoTryptic="isTryptic", nameFT="Specific",
+                                     filterProteotryptic=FALSE,
+                                     infoProteotypic="isProteotypic",
+                                     nameProteotypic="True",
+                                     filterMissedCleaved=FALSE,
+                                     maxMissedCleave=2,
+                                     infoMissedCleave="NMissedCleavages"){
+    ## check input and data
```

Comment:
Pain in the ass but I guess you have to add  tests for the type and structure of your matrices here. The list expects 1 to 4 (I think) matrices. So you can test if all provided entries of the list are (i) a matrix (ii) have the correct form and return an error 
same for the quantifyMatrix